bookserver/routers/assessment.py

```diff
@@ -143,3 +166,355 @@ async def get_history(request: Request, request_data: HistoryRequest):
     ]
 
     return make_json_response(detail=res)
+
+
+# Used by :ref:`compareAnswers`
+@router.get("/getaggregateresults")
+async def getaggregateresults(request: Request, div_id: str, course_name: str):
+    question = div_id
+
+    if not request.state.user:
```

Comment:
If a user isn't logged in, then `request.state` is None, so this causes an exception. I'd suggest adding a helper func to check this, since I see it a lot in the code.

Reply:
`request.state` is always there.  If a user is not logged in the the .user will be None.